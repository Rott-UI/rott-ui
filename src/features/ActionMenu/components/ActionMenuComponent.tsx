// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
