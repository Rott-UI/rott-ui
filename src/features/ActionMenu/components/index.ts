export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
