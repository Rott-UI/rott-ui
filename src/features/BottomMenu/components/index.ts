export * from './BottomMenu'
export * from './BottomMenuItem'
