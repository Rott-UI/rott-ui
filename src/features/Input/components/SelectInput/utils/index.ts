export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
