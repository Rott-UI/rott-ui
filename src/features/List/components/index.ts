export * from './List'
export * from './FavoriteSwipeComponent'
