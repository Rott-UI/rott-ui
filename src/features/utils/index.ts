export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
