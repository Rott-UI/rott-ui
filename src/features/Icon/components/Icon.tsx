/* eslint-disable indent */
// React Imports
import React, {FC} from 'react'

// Component Imports
import * as Icons from '../../../constants/Icons'
import {Item} from '../../../components/Item'

// Package Imports
import {colourFromVariant, display} from '../../../utils'
import {IconProps} from '../models'

export const Icon: FC<IconProps> = ({
  name,
  width = 16,
  height = 16,
  variant = 'white',
  strokeWidth = 1,
  mode = 'stroke',
  noStroke,
  color,
  ...props
}) => {
  const IconComponent = Icons[name]

  if (!IconComponent) return null

  return (
    <Item {...props}>
      <IconComponent
        fill={mode === 'fill' ? color ?? colourFromVariant(variant) : 'transparent'}
        stroke={
          mode === 'stroke'
            ? color ?? colourFromVariant(variant)
            : color ?? colourFromVariant(variant)
        }
        strokeWidth={
          mode === 'fill' && noStroke
            ? 0
            : mode === 'fill'
              ? 0.5
              : mode === 'stroke' && noStroke
                ? 0
                : strokeWidth
        }
        strokeLinecap='round'
        strokeLinejoin='round'
        width={display.px(width)}
        height={display.px(height)}
      />
    </Item>
  )
}
