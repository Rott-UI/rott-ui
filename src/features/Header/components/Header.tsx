// React Imports
import {FC, isValidElement} from 'react'

// Feature Imports
import {Item} from '../../Item'
import {Icon, IconProps} from '../../Icon'
import {Label} from '../../Label'
import {Image} from '../../Image'
import {HeaderProps} from '../models'
import {Content} from '../../Content'
import {Pressable} from '../../Pressable'
import {themeConfig} from '../../../providers'

// Provider Imports

/**
 *
 * @param title - Eğer Icon yerine bir title gözükmesi isteniyorsa bu property kullanılabilir.
 * @param subTitle - Alt Baslik.
 * @param back - Bir önceki sayfaya dönülmek istendiğinde bu property'nin verilmesi yeterli olacaktır. Historyde kayıtlı geridönülebilir sayfa varsa otomatikman o sayfaya gidecektir.
 * @param logo - Verilen logo header'ın ortasında renderlanacaktır.
 * @param rightElement - Sağ tarafta render olacak elementi verebilirsiniz
 * @param rightIcon - Icon setlerinden herhangi birini verirseniz ekranda renderlanır
 * @param rightIconOnPress - Sağ tarafta renderlanan icon'a basıldığından gerçekleşmesi gereken bir işlem varsa bu prop kullanılabilir.
 * @param leftElement - Sol tarafta render olacak elementi verebilirsiniz
 * @param leftIcon - Icon setlerinden herhangi birini verirseniz ekranda renderlanır
 * @param leftIconOnPress - Sol tarafta renderlanan icon'a basıldığından gerçekleşmesi gereken bir işlem varsa bu prop kullanılabilir.
 * @param casal - Arka planın casal renkte olmasını istediğinizde kullanabilirsiniz.
 * @param paddingHorizontal - Yatayda kisisellestirilebilir bosluk
 * @param paddingVertical - Dikeyde kisisellestirilebilir bosluk
 *
 */
export const Header: FC<HeaderProps> = ({
  back,
  title,
  subTitle,
  logo,
  leftIcon,
  leftElement,
  rightIcon,
  rightElement,
  defaultBackgroundColor,
  backgroundColor,
  paddingHorizontal,
  paddingVertical,
  goBackFunction,
  children,
  height = 56,
  ...props
}) => {
  return (
    <Content
      testID='header-test-id'
      minHeight={height}
      noPadding
      paddingHorizontal={paddingHorizontal ?? 24}
      paddingVertical={paddingVertical ?? 0}
      backgroundColor={defaultBackgroundColor ? themeConfig.colors['grey-800'] : backgroundColor}
      justifyContentCenter
      alignItemsCenter
      {...props}>
      <Item row height={height}>
        {/* Left Side */}
        <Item relative flexShrink={1} height={height} minWidth={height} justifyContentCenter>
          {!leftElement && !isValidElement(leftElement) && (
            <Pressable
              {...(typeof leftIcon === 'object' ? {...leftIcon} : null)}
              testID='header-left-pressable-test-id'
              width={height}
              height={height}
              minWidth={height}
              alignItemsFlexStart={!leftIcon?.alignItemsCenter}
              alignItemsCenter={!!leftIcon?.alignItemsCenter}
              backgroundColor={leftIcon?.backgroundColor}
              borderRadius={leftIcon?.rounded ? height : leftIcon?.borderRadius}
              justifyContentCenter
              onPress={(event) => {
                !!leftIcon?.onPress && leftIcon?.onPress(event)
                back && !!goBackFunction && goBackFunction()
              }}>
              <Icon
                testID='header-left-icon-test-id'
                width={24}
                height={24}
                strokeWidth={back ? 2 : leftIcon?.strokeWidth}
                variant={(leftIcon as IconProps)?.variant}
                mode={(leftIcon as IconProps)?.mode ?? 'stroke'}
                noStroke={
                  (leftIcon as IconProps)?.noStroke ?? (leftIcon as IconProps)?.mode === 'fill'
                }
                {...(typeof leftIcon === 'object' ? (leftIcon as IconProps) : {})}
                name={back ? 'CHEVRON_LEFT' : (leftIcon as IconProps)?.name ?? leftIcon}
              />
            </Pressable>
          )}

          {leftElement && isValidElement(leftElement) && leftElement}
        </Item>
        {/* Left Side */}

        {/* Middle */}
        <Item
          relative
          flex={1}
          height={height}
          minWidth={height}
          // backgroundColor='red'
          justifyContentCenter
          alignItemsCenter>
          {logo && <Image testID='header-logo-test-id' width={127} height={24} name={logo} />}

          {!logo && title && (
            <Label
              testID='header-title-test-id'
              textCenter
              fontSize='lg'
              fontFamily='Markpro-Bold'
              justifyContentCenter
              variant='white'
              numberOfLines={1}
              {...(typeof title === 'object' ? {...title} : null)}>
              {typeof title === 'string' ? title : title.text}
            </Label>
          )}
          {!logo && subTitle && (
            <Label
              testID='header-subtitle-test-id'
              textCenter
              fontSize='md'
              justifyContentCenter
              fontFamily='Markpro-Bold'
              variant='white'
              numberOfLines={1}
              {...(typeof subTitle === 'object' ? {...subTitle} : null)}>
              {typeof subTitle === 'string' ? subTitle : subTitle.text}
            </Label>
          )}
        </Item>
        {/* Middle */}

        {/* Right Side */}
        <Item
          relative
          flexShrink={1}
          height={height}
          minWidth={height}
          justifyContentCenter
          alignItemsFlexEnd>
          <Pressable
            {...(typeof rightIcon === 'object' ? {...rightIcon} : null)}
            testID='header-right-pressable-test-id'
            height={height}
            minWidth={height}
            alignItemsFlexStart={!rightIcon?.alignItemsCenter}
            alignItemsCenter={!!rightIcon?.alignItemsCenter}
            backgroundColor={rightIcon?.backgroundColor}
            borderRadius={leftIcon?.rounded ? height : leftIcon?.borderRadius}
            justifyContentCenter
            alignItemsFlexEnd
            onPress={(event) => !!rightIcon?.onPress && rightIcon?.onPress(event)}>
            {!rightElement && !isValidElement(rightElement) && (
              <Icon
                testID='header-right-icon-test-id'
                width={24}
                height={24}
                strokeWidth={back ? 2 : rightIcon?.strokeWidth}
                variant={(rightIcon as IconProps)?.variant}
                mode={(rightIcon as IconProps)?.mode ?? 'stroke'}
                noStroke={
                  (rightIcon as IconProps)?.noStroke ?? (rightIcon as IconProps)?.mode === 'fill'
                }
                {...(typeof rightIcon === 'object' ? (rightIcon as IconProps) : {})}
                name={(rightIcon as IconProps)?.name ?? rightIcon}
              />
            )}
          </Pressable>

          {rightElement && isValidElement(rightElement) && rightElement}
        </Item>
        {/* Right Side */}
      </Item>

      {children}
    </Content>
  )
}
