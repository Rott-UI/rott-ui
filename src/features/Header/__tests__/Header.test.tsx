//Component Imports
import {Header} from '../components'
import {Icon, IconTypes} from '../../Icon'

//Util and Lib Imports
import {act, fireEvent, render, waitFor} from './../../../utils'

describe('Header -> Custom Component', () => {
  const testId = {
    component: 'header-test-id',
    title: 'header-title-test-id',
    logo: 'header-logo-test-id',
    leftIcon: 'header-left-icon-test-id',
    leftIconPressable: 'header-left-pressable-test-id',
    rightIconPressable: 'header-right-pressable-test-id',
    rightIcon: 'header-right-icon-test-id',
    back: 'header-back-test-id',
  }

  const IMG: {logo: IconTypes; rightIcon: IconTypes} = {
    logo: 'HGS',
    rightIcon: 'REMOVE_CIRCLE',
  }
  const headerText = 'Test Başlık'

  it('yalnızca başlık çalışmalı ve snapshot eşleşmeli', () => {
    const {component} = testId

    const rendered = render(<Header testID={component} title={headerText} />)

    expect(rendered).toMatchSnapshot()
  })

  it('verilen testId ekranda olmalı', () => {
    const {title, rightIcon, logo, component} = testId
    const {queryByTestId} = render(<Header testID={component} title={headerText} />)

    //Title ekranda olmalı
    expect(queryByTestId(title)).toBeOnTheScreen()

    //RightIcon ekranda olmamalı
    expect(queryByTestId(rightIcon)).not.toBeOnTheScreen()

    //Logo ekranda olmamalı
    expect(queryByTestId(logo)).not.toBeOnTheScreen()
  })

  it('yalnızca logo çalışmalı ve snapshot eşleşmeli', () => {
    const {title, rightIcon, logo, component} = testId
    const {queryByTestId} = render(<Header testID={component} logo={IMG.logo} />)

    const logoElement = queryByTestId(logo)
    const rightIconElement = queryByTestId(rightIcon)
    const titleElement = queryByTestId(title)

    expect(logoElement).toBeOnTheScreen()
    expect(rightIconElement).not.toBeOnTheScreen()
    expect(titleElement).not.toBeOnTheScreen()
  })

  it('başlık ve icon var ise sadece icon gorunmeli', () => {
    const {rightIcon, logo, component} = testId
    const {queryByTestId, getByTestId} = render(
      <Header testID={component} title={headerText} logo={IMG.logo} />
    )

    const headerElement = getByTestId(component)
    const logoElement = getByTestId(logo)
    const rightIconElement = queryByTestId(rightIcon)

    expect(rightIconElement).not.toBeOnTheScreen()
    expect(logoElement).toBeOnTheScreen()
    expect(headerElement).not.toHaveTextContent(headerText)
  })

  it('rightIcon görüntülenmeli ve onPress Methodu Çalışmalı', async () => {
    const {component, rightIcon: rightIconTestId, rightIconPressable} = testId
    const {rightIcon} = IMG

    const mockOnPress = jest.fn()
    const {getByTestId} = render(
      <Header
        testID={component}
        title={headerText}
        rightIcon={{
          testID: rightIconTestId,
          name: rightIcon,
          onPress: mockOnPress,
        }}
      />
    )

    const rightIconElement = getByTestId(rightIconTestId)
    const rightIconPressableElement = getByTestId(rightIconPressable)
    //RightIcon On Press
    await act(async () => {
      await waitFor(() => {
        fireEvent.press(rightIconPressableElement!)
      })
    })

    expect(rightIconElement).toBeOnTheScreen()
    expect(mockOnPress).toHaveBeenCalled()
  })

  it('rightIcon verilmediğinde ekranda gözükmemeli', () => {
    const {component, rightIcon: rightIconTestId} = testId
    const {queryByTestId} = render(<Header testID={component} title={headerText} />)

    const rightIconElement = queryByTestId(rightIconTestId)

    expect(rightIconElement).not.toBeOnTheScreen()
  })

  it('rightElement görüntülenmeli', () => {
    const {component, rightIcon: rightIconTestId} = testId
    const {rightIcon} = IMG

    const {queryByTestId} = render(
      <Header
        testID={component}
        title={headerText}
        rightElement={<Icon testID={rightIconTestId} name={rightIcon} />}
      />
    )

    const rightIconElement = queryByTestId(rightIconTestId)

    expect(rightIconElement).toBeOnTheScreen()
  })

  it('leftIcon görüntülenmeli ve onPress Methodu Çalışmalı', async () => {
    const {component, leftIcon: leftIconTestId, leftIconPressable} = testId
    const {rightIcon} = IMG

    const mockOnPress = jest.fn()
    const {getByTestId} = render(
      <Header
        testID={component}
        title={headerText}
        leftIcon={{
          testID: leftIconTestId,
          name: rightIcon,
          onPress: mockOnPress,
        }}
      />
    )

    const leftIconElement = getByTestId(leftIconPressable)
    const leftIconPressableElement = getByTestId(leftIconPressable)
    //LeftIcon On Press
    await act(async () => {
      await waitFor(() => {
        fireEvent.press(leftIconPressableElement!)
      })
    })

    expect(leftIconElement).toBeOnTheScreen()
    expect(mockOnPress).toHaveBeenCalled()
  })

  it('leftIcon verilmediğinde ekranda gözükmemeli', () => {
    const {component, leftIcon: leftIconTestId} = testId
    const {queryByTestId} = render(<Header testID={component} title={headerText} />)

    const leftIconElement = queryByTestId(leftIconTestId)

    expect(leftIconElement).not.toBeOnTheScreen()
  })

  it('leftElement görüntülenmeli', () => {
    const {component, rightIcon: rightIconTestId} = testId
    const {rightIcon} = IMG

    const {queryByTestId} = render(
      <Header
        testID={component}
        title={headerText}
        rightElement={<Icon testID={rightIconTestId} name={rightIcon} />}
      />
    )

    const rightIconElement = queryByTestId(rightIconTestId)

    expect(rightIconElement).toBeOnTheScreen()
  })

  it('rightIcon ve leftIcon verildiğinde, ekranda gözükmeli', () => {
    const {component, rightIcon: rightIconTestId, leftIcon: leftIconTestId} = testId
    const {rightIcon} = IMG

    const {queryByTestId} = render(
      <Header
        testID={component}
        title={headerText}
        leftIcon={{
          testID: leftIconTestId,
          name: rightIcon,
        }}
        rightIcon={{
          testID: rightIconTestId,
          name: rightIcon,
        }}
      />
    )

    const rightIconElement = queryByTestId(rightIconTestId)
    const leftIconElement = queryByTestId(leftIconTestId)

    expect(rightIconElement).toBeOnTheScreen()
    expect(leftIconElement).toBeOnTheScreen()
  })

  it('children verildiğinde, ekranda gözükmeli', () => {
    const {component} = testId
    const {queryByTestId} = render(
      <Header testID={component} title={headerText}>
        <Icon testID='header-children-icon-test-id' name='HGS' />
      </Header>
    )

    const childrenElement = queryByTestId('header-children-icon-test-id')

    expect(childrenElement).toBeOnTheScreen()
  })
})
