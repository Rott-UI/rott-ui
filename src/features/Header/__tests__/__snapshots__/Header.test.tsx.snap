// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header -> Custom Component yalnızca başlık çalışmalı ve snapshot eşleşmeli 1`] = `
<View
  minHeight={56}
  paddingVertical={0}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderBottomEndRadius": undefined,
      "borderBottomStartRadius": undefined,
      "borderRadius": undefined,
      "borderTopEndRadius": undefined,
      "borderTopStartRadius": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": "center",
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": 108,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": 46,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": 0,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
  testID="header-test-id"
>
  <View
    height={56}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "flexGrow": undefined,
        "gap": undefined,
        "height": 108,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": 108,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
  >
    <View
      flexShrink={1}
      height={56}
      justifyContentCenter={true}
      minWidth={56}
      relative={true}
      style={
        {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": 108,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 108,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": 108,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": "relative",
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        alignItemsCenter={false}
        alignItemsFlexStart={true}
        animated={false}
        collapsable={false}
        focusable={true}
        height={56}
        justifyContentCenter={true}
        minWidth={56}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "flex-start",
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": 108,
            "justifyContent": "center",
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": 108,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": 108,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": 108,
            "zIndex": undefined,
          }
        }
        testID="header-left-pressable-test-id"
        width={56}
      />
    </View>
    <View
      alignItemsCenter={true}
      flex={1}
      height={56}
      justifyContentCenter={true}
      minWidth={56}
      relative={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": 1,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": 108,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 108,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": 108,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": "relative",
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <Text
        justifyContentCenter={true}
        numberOfLines={1}
        style={
          {
            "backgroundColor": undefined,
            "bottom": undefined,
            "color": "rgba(255, 255, 255, 1)",
            "flexGrow": undefined,
            "fontFamily": "Markpro-Bold",
            "fontSize": 16,
            "fontWeight": undefined,
            "gap": undefined,
            "height": undefined,
            "left": undefined,
            "letterSpacing": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "textAlign": "center",
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
        testID="header-title-test-id"
      >
        Test Başlık
      </Text>
    </View>
    <View
      alignItemsFlexEnd={true}
      flexShrink={1}
      height={56}
      justifyContentCenter={true}
      minWidth={56}
      relative={true}
      style={
        {
          "alignItems": "flex-end",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": 108,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 108,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": 108,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": "relative",
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        alignItemsCenter={false}
        alignItemsFlexEnd={true}
        alignItemsFlexStart={true}
        animated={false}
        collapsable={false}
        focusable={true}
        height={56}
        justifyContentCenter={true}
        minWidth={56}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "flex-end",
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": 108,
            "justifyContent": "center",
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": 108,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": 108,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
        testID="header-right-pressable-test-id"
      />
    </View>
  </View>
</View>
`;
